openassessment/xblock/static/js/src/lms/oa_base.js

```diff
@@ -73,6 +74,25 @@ OpenAssessment.BaseView.prototype = {
     },
 
     /**
+     * Install click handlers to set usage key of an XBlock.
+     */
+    setUpUsageID: function() {
+        $(".action--submit").on("click", function(event) {
+            this.usageID = $(event.target).closest('.xblock-student_view-openassessment').data('usage-id');
+        });
+    },
+
+    /**
+     * Get usage key of an XBlock.
+     */
+    getUsageID: function() {
+        if (!this.usageID) {
+            this.usageID = $(this.element).data('usage-id');
```

Comment:
Oh, nice question. You are right. I had to go in to deep to answer your question. 

I think, we can just go with `getUsageID` method. It is cleaner and better approach. Whenever we need `usageId` in any ORA step (self, response, peer, staff grade etc.), we can use `getUsageID` method to het the usageID on demand as I have done for peer step.

Initially, I was going with click handler option, then I was writing jasmine tests and I had to create `getUsageID` method. But your question made me to realise, do we need both. We can go with `getUsageID`.